### ISSUE TITLE
mc diff ./ ./ returns wrong result

### ISSUE THREAD

**Issue by vadmeste** [minio/mc#1674]
```
$ mkdir /tmp/foo/ ; cd /tmp/foo; touch file ; mc diff ./ ./ 
file and ./file - only in first.
```

At the first glance, it seems that the problem is in https://github.com/minio/mc/blob/master/difference.go#L63 . Here difference.go is getting `file` from source.List() & target.List() without any leading ./ but decided to add targetURL (./) in only one of the names before comparing them. It is sufficient that we make difference.go smarter when comparing paths to correct the problem, however I feel that the problem is in client.List() itself which is returning file names without any leading ./ in this case.

What does client.List() return ?

I feel that there is an inconsistency here, client.List() could return:
*) /bucket/file
*)../file
*) file

Also `..` is not allowed in S3, and it is better to drop it to have a complete consistency between fs and s3 clients, to avoid any further complication later 

What I would suggest is that you make client.List() really dumb, it returns the list of files and dirs names inside the target directory without any leading path in the returned urls.

**Comment by harshavardhana** [comment]
Is this bug fixed with #1681 ?

**Comment by vadmeste** [comment]
yes
